Admin/ArticleAdmin.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\ArticleBundle\Admin;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Datagrid\ListMapper;
+use Sonata\AdminBundle\Form\FormMapper;
+use Sonata\AdminBundle\Show\ShowMapper;
+use Sonata\ArticleBundle\Model\Article;
+use Sonata\ArticleBundle\Model\Fragment;
+use Sonata\CoreBundle\Validator\ErrorElement;
+use Sonata\MediaBundle\Admin\ORM\MediaAdmin;
+
+/**
+ * @author Florent Denis <florent.denis@ekino.com>
+ */
+class ArticleAdmin extends AbstractAdmin
```

Comment:
It's depend of the use case IMHO.

I think we can make it `final` right now as it's specific.

If a user want to extends it, he cans create an issue. Then we can estimate if removing the `final` keyword make sens or if this can be improved by configuration.

The reverse (no final to final) is harden to do because of BC break handling.